### PR TITLE
Port algorithms from the JTS Angle class

### PR DESCRIPTION
**Pull request by jesseeichar**
This PR builds on the XYZ PR. So review and merge that one first

### CODE REVIEW

File: xy/angle.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	"github.com/twpayne/go-geom/xy/orientation"
+	"math"
+)
+
+const piTimes2 = math.Pi * 2
+
+// ToDegrees converts from radians to degrees.
+//
+// Param radians - an angle in radians
+func ToDegrees(radians float64) float64 {
```
Comment (twpayne) [inline xy/angle.go]: `ToDegrees` and `ToRadians` are so simple that they shouldn't be exported.

File: xy/angle.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	return (radians * 180) / (math.Pi)
```
Comment (twpayne) [inline xy/angle.go]: No brackets are needed here. I'm surprised that `go fmt` didn't fix this.

File: xy/angle.go [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+func AngleOrientation(ang1, ang2 float64) orientation.Type {
+	crossproduct := math.Sin(ang2 - ang1)
+
+	if crossproduct > 0 {
+		return orientation.CounterClockwise
+	}
+	if crossproduct < 0 {
+		return orientation.Clockwise
+	}
+	return orientation.Collinear
```
Comment (twpayne) [inline xy/angle.go]: Consider using a `switch` statement here, something like:
```go
switch {
case crossproduct > 0:
    // ...
case crossproduct < 0:
    // ...
default:
    // ...
}
```

**Comment by twpayne** [comment]
Looks good! Please merge. We can tidy up the formatting (to improve consistency) in a separate PR.

**Comment by jesseeichar** [comment]
I agreed with all comments and updated the PR to have these changes.

**Pull request merged**